### DISCUSSION TITLE
Audio device x Windows Server

### DISCUSSION THREAD

**Discussion by pedroalpenna** [sipsorcery-org/sipsorcery#442] [Q&A]
Hi all,

Recently, We're learning the sipsorcery. We've made some tests with an application and it works well in our local machine. 

However, when we run this same application in a Windows Server 2019, an error message appears: "Must call init first".

How would it be possible to run the program on the windows server since it is necessary to have an audio device? Is it really necessary to have it? Has anyone ever experienced this?

Best regards!

**Comment by sipsorcery** [comment]
You don't need an audio device unless you want to capture or playback audio. It's also easy to check if a particular demo needs a Windows audio device by checking the project file for a `SIPSorceryMedia.Windows` dependency.

I'd recommend having a look at the [WebRTC Test Pattern Server demo](https://github.com/sipsorcery-org/sipsorcery/tree/master/examples/WebRTCExamples/WebRTCTestPatternServer). It doesn't use an audio track so should work fine on Windows Server.

**Reply by pedroalpenna** [reply]
Thanks for the answer, the main idea of our project is to play an audio. For example, if the user calls 1111 from his device, another device will receive a call followed by an automatic answer and then the sound is played.

It is possible to do it from windows server? As I said, from a local machine it works fine.

**Comment by sipsorcery** [comment]
Whoops I assumed you were doing a WebRTC app. Seems like you're doing a VoIP one instead.

The answer is still yes, you don't need an audio device to source or pass though audio streams.

The [SIPCallServer](https://github.com/sipsorcery-org/sipsorcery/blob/master/examples/SIPExamples/SIPCallServer/Program.cs) demo is a minimal SIP server that doesn't use an audio device. If you start it you can get different audio signals depending on the incoming URI. For example calling `0@192.168.0.1` (replace with your IP address) should get you the on-hold music.
